app/src/main/java/org/piwigo/ui/main/MainActivity.java

```diff
@@ -343,31 +409,37 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (requestCode == SELECT_PICTURES) {
             if (resultCode == RESULT_OK) {
-                if (data.getData() != null) {
-
-                    String imageName = "";
-                    Uri targetUri = data.getData();
-                    if (data.toString().contains("content:")) {
-                        imageName = getRealPathFromURI(targetUri);
-                    } else if (data.toString().contains("file:")) {
-                        imageName = targetUri.getPath();
-                    } else {
-                        imageName = null;
-                        /* TODO add proper error handling */
-                    }
-
+                if (data.getClipData() != null) {
                     Intent intent = new Intent(this, UploadService.class);
-                    intent.putExtra(UploadService.KEY_IMAGE_NAME, imageName);
-                    intent.putExtra(UploadService.KEY_IMAGE_URI, targetUri);
-                    intent.putExtra(UploadService.KEY_ACCOUNT, userManager.getActiveAccount().getValue());
-                    intent.putExtra(UploadService.KEY_CATEGORY_ID, getCurrentCategoryId());
-
+                    ImageUploadQueue<UploadAction> imageUploadQueue = new ImageUploadQueue<>();
```

Comment:
shouldn't the queue be located in the UploadService? Because finally it should be the same whether I upload two files directly after each other (second before the first is completely uploaded) or directly two pictures in one selection